code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/ServiceLinkPreInstanceStateHandler.java

```diff
@@ -35,7 +35,7 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
         Set<InstanceLink> links = new HashSet<>();
         Set<InstanceLink> toRemove = new HashSet<>();
 
-        for (ServiceConsumeMap map : consumeMapDao.findConsumedServicesForInstance(instance.getId()) ) {
+        for (ServiceConsumeMap map : consumeMapDao.findConsumedServicesForInstance(instance.getId(), "service") ) {
```

Comment:
Minor: we can put ServiceDiscoveryConstants.KIND.SERVICE instead of "service" here